File: Kaskus/CueText/Dede.App.CueTextVb/Modules/Extensions.vb
Imports System.Windows.Forms
Imports System.Runtime.CompilerServices
Imports System.Runtime.InteropServices

Namespace Modules
    Public Module Extensions

        <Extension> _
        Public Sub CueText(control As Control, text As [String])
            NativeMethods.SendMessageW(If(TypeOf control Is ComboBox, control.GetComboBoxInfo().hwndItem, control.Handle), NativeMethods.EmSetCueBanner, CType(0, IntPtr), text)
        End Sub

        <Extension> _
        Friend Function GetComboBoxInfo(control As IWin32Window) As NativeMethods.ComboboxInfo
            Dim info = New NativeMethods.ComboboxInfo()
            info.cbSize = Marshal.SizeOf(info)
            NativeMethods.GetComboBoxInfo(control.Handle, info)
            Return info
        End Function
    End Module
End Namespace

File: Kaskus/CueText/CueTextCs/Extensions.cs
namespace Dede.App.CueText
{
    using System;
    using System.Runtime.InteropServices;
    using System.Windows.Forms;

    public static class Extensions
    {
        public static void CueText(
            this Control control,
            String text)
        {
            NativeMethods.SendMessageW(
                control is ComboBox
                     ? control.GetComboBoxInfo().hwndItem
                     : control.Handle,
                NativeMethods.EmSetCueBanner,
                (IntPtr)0,
                text);
        }

        internal static NativeMethods.ComboboxInfo GetComboBoxInfo(
            this IWin32Window control)
        {
            var info = new NativeMethods.ComboboxInfo();
            info.cbSize = Marshal.SizeOf(info);
            NativeMethods.GetComboBoxInfo(control.Handle, ref info);
            return info;
        }
    }
}